brave-cxf3/src/main/java/com/github/kristofa/brave/cxf3/BraveAsyncJaxRsThreadCleaner.java

```diff
@@ -0,0 +1,44 @@
+package com.github.kristofa.brave.cxf3;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.ServerSpanThreadBinder;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.Context;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+
+/**
+ * Cleans current server thread binder in async scenarios.
+ */
+class BraveAsyncJaxRsThreadCleaner implements ContainerRequestFilter {
```

Comment:
It could do it, but I wanted clean dependency. If someone uses only jaxws, jaxrs dependency would be excessive. But we can merge those two if it's not a problem.